src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,61 @@
+import classNames from "classnames"
+import styles from "./ProgressBar.module.css"
+
+const MIN = 0
+const MAX = 100
+
+const getTransformStyle = (value?: number) => {
+  if (value || value === MIN) {
+    const transform = value - MAX
+    return `translateX(${transform}%)`
+  } else {
+    return
+  }
+}
+
+const getIndicatorClasses = (value?: number) => {
+  return classNames(styles.indicator, {
+    [styles.indeterminate]: value !== MIN && !value,
+  })
+}
+
+const getAriaProgressLabels = (value?: number) => {
+  if (value || value === MIN) {
+    return {
+      "aria-valuenow": value,
+      "aria-valuemin": MIN,
+      "aria-valuemax": MAX,
+    }
+  } else {
+    return {}
+  }
+}
+
+export interface ProgressBarProps {
+  /** A number between 0-100 */
+  value?: number
+  ariaLabel?: string
+}
+
+export function ProgressBar({ value, ariaLabel }: ProgressBarProps) {
+  const indicatorClasses = getIndicatorClasses(value)
+  const transform = getTransformStyle(value)
+  const ariaLabels = getAriaProgressLabels(value)
```

Comment:
This line could then look like this instead:
```tsx
const ariaLabels = value ? getAriaProgressLabels(value) : {};
```

The benefit is that you see all the possible branching logic here in the component instead of hidden inside internal operators.

This is an idea that sometimes bubbles up from the function level to the component level and can reveal problems where there's no defined behavior for the cases where the values don't exist.